agent/structs/config_entry_gateways.go

```diff
@@ -203,6 +204,34 @@ func validateHost(tlsEnabled bool, host string) error {
 	return nil
 }
 
+func (e *IngressGatewayConfigEntry) ListRelatedServices() []ServiceID {
```

Comment:
FYI this is the same mechanism that `service-resolver`, `service-splitter`, and `service-router` use during graph validation. It is indexed on the config entry memdb table and named `"link"` (which is only used in one place).

Reply:
I would probably at least add a comment here that specifies that it will not return services that are matched by the wildcard specifier. That could be potentially confusing.

Reply:
done